Store and load support for GPTConfig to json

djlisbonne opened the pull request:
Added support to create new GPTConfig object from pre-saved json file, and support to save an existing GPTConfig object to json for later model creation

mmoffatt2 commented:
Can we add a train.py argparse arg to allow loading a json file at the start of training?

djlisbonne commented:
Definitely, that's a good suggestion. Made some changes: 

- I added a flag called "--load_config_json" that accepts a json filename string, and then loads all params (model, training and logging) directly from the file. **Note**: it updates params after the defaults have been set, so any parameter not explicitly defined in the provided json will be set to its default value.
- I also added a "--save_config_json" that will save all params (model, training and logging) to a json file. 
- Changed both flags to be within the logging_group so they don't appear in the model parameters

I'm keeping the two functions in GPTConfig, "from_json" and "to_json" (renamed), so that if a user wants to only store the model params, that's still possible without having to run a long train.py command from the terminal.

gkielian commented:
This sounds great!

This helps guard against the scenario where we change default values.
